can/can.cpp

```diff
@@ -0,0 +1,255 @@
+    if ((filterMode != CAN_FilterMode::PATH_ALL_ID) && (filterMode != CAN_FilterMode::PATH_ONE_TYPE_STD_OR_EXT_ID_GROUP)) {
+        return CAN_ClassSettingStatus::WRONG_SET_PATH_ID_FUNCTION_USED;
+    } else if (filterMode == CAN_FilterMode::PATH_ALL_ID) {
+        return CAN_ClassSettingStatus::ALL_ID_PATH;
+    }
+
+    std::array<uint8_t, 2> bitShift;
+    if (idType == CAN_ID_STD) {
+        bitShift[0] = 21;
+    } else if (idType == CAN_ID_EXT) {
+        bitShift[0] = 3;
+    }
+
+    filterId[0] = minId << bitShift[0];
+
+    minId <<= 21;
+    maxId <<= 21;
+    filterMask[0] = ((~(minId ^ maxId)) >> 21);
+    filterMask[0] <<= bitShift[0];
+
+    filterConfig.FilterIdHigh = filterId[0] >> 16;
+    filterConfig.FilterIdLow = filterId[0];
+
+    filterConfig.FilterMaskIdHigh = filterMask[0] >> 16;
+    filterConfig.FilterMaskIdLow = filterMask[0];
+
+    return CAN_ClassSettingStatus::NON_ERROR;
+}
+
+CAN_StatusType Can::setFilterConfig() {
+    return static_cast<CAN_StatusType>(HAL_CAN_ConfigFilter(hcan, &filterConfig));
+}
+
+void Can::setId(uint32_t idType, uint32_t id) {
+    txHeader.IDE = idType;
+
+    if (txHeader.IDE == CAN_ID_STD) {
+        txHeader.StdId = id;
+    } else if (txHeader.IDE == CAN_ID_EXT) {
+        txHeader.ExtId = id;
+    }
+}
+
+void Can::setDataFrame(uint32_t dataFrameType) {
+    txHeader.RTR = dataFrameType;
+}
+
+bool Can::isMailBoxPending(uint32_t txMailbox) {
+    if (HAL_CAN_IsTxMessagePending(hcan, txMailbox) != 0) {
+        return true;
+    }
+    return false;
+}
+
+CAN_StatusType Can::transmit(uint8_t dataLength, uint8_t txData[]) {
+    txHeader.DLC = dataLength;
+    return static_cast<CAN_StatusType>(HAL_CAN_AddTxMessage(hcan, &txHeader, txData, &usedTxMailbox));
+}
+
+uint32_t Can::getUsedTxMailbox() {
+    return usedTxMailbox;
+}
+
+CAN_StatusType Can::receive(uint32_t rxFifo, uint8_t rxData[]) {
+    if (HAL_CAN_GetRxFifoFillLevel(hcan, rxFifo) == 0) {
+        return CAN_StatusType::CAN_Rx_FIFO_Empty;
+    }
+    return static_cast<CAN_StatusType>(HAL_CAN_GetRxMessage(hcan, rxFifo, &rxHeader, rxData));
+}
+
+uint32_t Can::getRxIdType() {
+    return rxHeader.IDE;
+}
+
+uint32_t Can::getRxId() {
+    if (rxHeader.IDE == CAN_ID_STD) {
+        return rxHeader.StdId;
+    } else if (rxHeader.IDE == CAN_ID_EXT) {
+        return rxHeader.ExtId;
+    } else {
+        return static_cast<uint32_t>(CAN_StatusType::HAL_ERROR);
+    }
+}
+
+uint32_t Can::getRxDataLength() {
+    return rxHeader.DLC;
+}
+
+CAN_StatusType Can::abortTransmit(uint32_t txMailBox) {
+    if (HAL_CAN_IsTxMessagePending(hcan, txMailBox) != 1) {
+        return CAN_StatusType::CAN_Tx_Not_Pending;
+    }
+    return static_cast<CAN_StatusType>(HAL_CAN_AbortTxRequest(hcan, txMailBox));
+}
+
+HAL_CAN_StateTypeDef Can::getState() {
+    return hcan->State;
+}
+
+uint32_t Can::getFilterId(uint8_t idNum) {
+    return filterId[idNum];
+}
+
+uint32_t Can::getMaskId(uint8_t maskNum) {
+    return filterMask[maskNum];
+}
```

Comment:
[SHOULD] 変数名・もしくは関数名に名前をそろえたいです。

---

```suggestion
uint32_t Can::getMaskId(uint8_t index) {
    return filterMask[index];
}
```

[WANT] index の方が適切だと思います

Reply:
bac9fe61810e1a8672e7f552cda80e15a306e74c で削除